framework/src/actions/MooseObjectAction.C

```diff
@@ -27,6 +28,9 @@ validParams<MooseObjectAction>()
   return params;
 }
 
+template <>
+InputParameters validParams<MooseObject>();
```

Comment:
```
template <>
InputParameters validParams<MooseObjectAction>()
{
  InputParameters params = validParams<Action>();
  return params;
}
```
?

Reply:
Oh, the body is on lines 20-29. But why do we need this forward declaration here?